### PR TITLE
Service Bus API implementation

### PR DESCRIPTION
**Pull request by sjuarezgx**
Issue:97173

### CODE REVIEW

File: dotnet/src/dotnetcore/Providers/Messaging/GXMessageBroker/MessageBrokerProvider.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+		}
+
+		public MessageQueue Connect(string providerTypeName, GXProperties properties, out GXBaseCollection<SdtMessages_Message> errorMessages, out bool success)
+		{
+			errorMessages = new GXBaseCollection<SdtMessages_Message>();
+			MessageQueue messageQueue = new MessageQueue();
+			if (string.IsNullOrEmpty(providerTypeName))
+			{
+				GXUtil.ErrorToMessages("GXMessageBroker1000", "Message Broker provider cannot be empty", errorMessages);
```
Comment (ggallotti) [inline dotnet/src/dotnetcore/Providers/Messaging/GXMessageBroker/MessageBrokerProvider.cs:27]: "'GXMessageBroker1000". Why the "1000"?

File: dotnet/src/dotnetcore/Providers/Messaging/GXAzureServiceBus/ServiceBusMessageBrokerProvider.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+		public MessageQueue Connect(string queueName, string connectionString, bool sessionEnabled, GxUserType receiverOptions, string senderIdentifier, out GXBaseCollection<SdtMessages_Message> errorMessages, out bool success)
+		{
+
+			MessageBrokerProvider messageBrokerProvider = new MessageBrokerProvider();
+			ReceiverOptions options = TransformGXUserTypeToReceiverOptions(receiverOptions);
+
+			GXProperties properties = new GXProperties();
+			properties.Add(PropertyConstants.MESSAGEBROKER_AZURESB_QUEUENAME, queueName);
+			properties.Add(PropertyConstants.MESSAGEBROKER_AZURESB_CONNECTIONSTRING, connectionString);
+			properties.Add(PropertyConstants.SESSION_ENABLED, sessionEnabled.ToString());
```
Comment (ggallotti) [inline dotnet/src/dotnetcore/Providers/Messaging/GXAzureServiceBus/ServiceBusMessageBrokerProvider.cs:45]: It seems that the Connect code is the same code in the 4 Connect methods. Maybe we can do a minor refactoring?

File: dotnet/src/dotnetcore/Providers/Messaging/GXAzureServiceBus/ServiceBusMessageBrokerProvider.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+			return messageQueue;
+		}
+
+		#region Transformation methods
+		private ReceiverOptions TransformGXUserTypeToReceiverOptions(GxUserType options)
+		{
+			ReceiverOptions receiverOptions = new ReceiverOptions();
+			if (options != null)
+			{ 
+				receiverOptions.ReceiveMode = options.GetPropertyValue<short>("Receivemode");
```
Comment (ggallotti) [inline dotnet/src/dotnetcore/Providers/Messaging/GXAzureServiceBus/ServiceBusMessageBrokerProvider.cs:86]: Receivemode is lowercase here. And camelcase on line 120: AddObjectProperty("ReceiveMode", ReceiveMode, false);
Maybe this is an error? IF not an error, ignore.

File: dotnet/src/dotnetcore/Providers/Messaging/GXMessageBroker/MessageBrokerProvider.cs (hunk 1 above)
Comment (sjuarezgx) [reply dotnet/src/dotnetcore/Providers/Messaging/GXMessageBroker/MessageBrokerProvider.cs:27]: You suggest starting with code 1? or what do you suggest? thanks

File: dotnet/src/dotnetcore/Providers/Messaging/GXAzureServiceBus/ServiceBusMessageBrokerProvider.cs (hunk 3 above)
Comment (sjuarezgx) [reply dotnet/src/dotnetcore/Providers/Messaging/GXAzureServiceBus/ServiceBusMessageBrokerProvider.cs:86]: No, it has no relation..

File: dotnet/src/dotnetcore/Providers/Messaging/GXAzureServiceBus/ServiceBusMessageBrokerProvider.cs (hunk 2 above)
Comment (sjuarezgx) [reply dotnet/src/dotnetcore/Providers/Messaging/GXAzureServiceBus/ServiceBusMessageBrokerProvider.cs:45]: It's not exactly the same in fact.. it has some differences that's why I couldn't use a common method..

**Comment by ggallotti** [comment]
I've added @claudiamurialdo to help for review this PR.

File: dotnet/src/dotnetcore/Providers/Messaging/GXMessageBroker/MessageBrokerProvider.cs (hunk 1 above)
Comment (sjuarezgx) [reply dotnet/src/dotnetcore/Providers/Messaging/GXMessageBroker/MessageBrokerProvider.cs]: I removed the numbers.

**Review by ggallotti — APPROVED** [review]

**Pull request merged**